Dockerfile

```diff
@@ -1,14 +1,40 @@
-FROM node:20-slim
+FROM node:20-slim as builder
 
 RUN apt-get update
-RUN apt-get install -y build-essential cmake git libssl-dev
+RUN apt-get install -y build-essential cmake git libssl-dev tini
 
 WORKDIR /app
 
-COPY . .
+COPY package*.json ./
+
 RUN npm ci --quiet
+
+COPY . .
+
 RUN npm run build
 
+ENV NODE_ENV production
+RUN npm prune --production
+
+FROM node:20-slim as app
+WORKDIR /app
+# built src without dev dependencies
+COPY --from=builder /app ./
+# tini is used to handle signals properly, see https://github.com/krallin/tini#using-tini
+COPY --from=builder /usr/bin/tini /usr/bin/tini
+
+# copy shared libraries (without having artifacts from apt-get install that is needed to build our application)
+COPY --from=builder /usr/lib/**/libcrypto*.so /usr/lib/
+COPY --from=builder /usr/lib/**/libssl*.so /usr/lib/
```

Comment:
```suggestion
COPY --from=builder /usr/lib/**/libcrypto* /usr/lib/
COPY --from=builder /usr/lib/**/libssl* /usr/lib/
```

Reply:
there is more than just the individual `*.so` files needed. It would be better to nail this down more, but removing the `.so` seems to fix the issue.